SnowplowTracker/SnowplowTracker/Storage/EventStore.cs

```diff
@@ -0,0 +1,296 @@
+/*
+ * EventStore.cs
+ * SnowplowTracker.Storage
+ * 
+ * Copyright (c) 2015 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ * 
+ * Authors: Joshua Beemster
+ * Copyright: Copyright (c) 2015 Snowplow Analytics Ltd
+ * License: Apache License Version 2.0
+ */
+
+using System;
+using System.Threading;
+using System.Runtime.CompilerServices;
+using System.Data;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Data.SqliteClient;
+using SnowplowTracker.Payloads;
+using SnowplowTracker.Collections;
+
+namespace SnowplowTracker.Storage {
+	public class EventStore {
+
+		// Database Details
+		private static readonly string DATABASE_NAME     = "URI=file:SnowplowEvents.db";
+		private static readonly string TABLE_NAME        = "events";
+		private static readonly string COLUMN_ID         = "id";
+		private static readonly string COLUMN_EVENT_DATA = "eventData";
+
+		// SQL Statements
+		private static readonly string CREATE_TABLE = "CREATE TABLE IF NOT EXISTS '" + TABLE_NAME + "' (" + COLUMN_ID + " INTEGER PRIMARY KEY, " + COLUMN_EVENT_DATA + " BLOB)";
+		private static readonly string INSERT_EVENT = "INSERT INTO " + TABLE_NAME + " (" + COLUMN_EVENT_DATA + ") " + "VALUES ";
+		private static readonly string COUNT = "SELECT COUNT(*) as events FROM " + TABLE_NAME;
+		private static readonly string SELECT_ALL = "SELECT * FROM " + TABLE_NAME;
+		private static readonly string SELECT_RANGE = "SELECT * FROM " + TABLE_NAME + " ORDER BY " + COLUMN_ID + " DESC LIMIT ";
+		private static readonly string SELECT_ROW = "SELECT * FROM " + TABLE_NAME + " WHERE " + COLUMN_ID + "=";
+		private static readonly string DELETE_ALL = "DELETE FROM " + TABLE_NAME;
+		private static readonly string DELETE_ROW = "DELETE FROM " + TABLE_NAME + " WHERE " + COLUMN_ID + "=";
+
+		// Connection
+		private IDbConnection dbConnection;
+
+		// Add Queue
+		private Thread payloadQueueConsumer;
+		private volatile bool payloadQueueConsumerContinue = true;
+		private ConcurrentQueue<TrackerPayload> payloadQueue = new ConcurrentQueue<TrackerPayload>();
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SnowplowTracker.Storage.EventStore"/> class.
+		/// </summary>
+		public EventStore() {
+			Open ();
+			payloadQueueConsumer = new Thread(new ThreadStart(PayloadQueueConsumer));
```

Comment:
I know that in Java you aren't meant to start a thread inside a constructor (e.g. [this SO question](http://stackoverflow.com/questions/5623285/why-not-to-start-a-thread-in-the-constructor-how-to-terminate)). Not sure to what extent the same is true of C# - there's a related question [here](http://stackoverflow.com/questions/7261173/c-sharp-start-async-method-within-object-constructor-bad-practice).